### PR TITLE
Feature/v4 new blendmode

### PR DESCRIPTION
**Pull request by NickyWeber**
Support for new blend mode of cocos2d v4. 
Cocos2d pull request for ccb binary reader is here: https://github.com/cocos2d/cocos2d-swift/pull/1202 
Added lots of tests for reader and writer.
Some refactoring.

### CODE REVIEW

**Comment by NickyWeber** [comment]
@cocojoe @rkachowski anyone?

File: SpriteBuilder/SpriteBuilder Tests/CCBDictionaryReader_Tests.m [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+    // "ccbFile" -> ""
+    // Cannot import CCBPCCBFile.h for whatever reason
+    CCNode *subCCB = node.children[0];
+    XCTAssertTrue([node.children[0] isKindOfClass:[NSClassFromString(@"CCBPCCBFile") class]]);
+
+    XCTAssertEqualObjects([subCCB.userObject extraProps][@"ccbFile"], @"test_ccbreader_sprite.ccb");
+}
+
+/*
+- (void)testNodeGraphFromDocumentDict_physics_distance_joint
```
Comment (rkachowski) [inline SpriteBuilder/SpriteBuilder Tests/CCBDictionaryReader_Tests.m:528]: why is this commented out?

File: SpriteBuilder/ccBuilder/AppDelegate.m [hunk 2]
```diff
@@ -1367,14 +1367,24 @@ - (void) replaceDocumentData:(NSMutableDictionary*)doc
...
+    }
+
     NSMutableArray* loadedJoints = [NSMutableArray array];
     if(doc[@"joints"] != nil)
     {
         for (NSDictionary * jointDict in doc[@"joints"])
         {
-            CCNode * joint = [CCBReaderInternal nodeGraphFromDictionary:jointDict parentSize:CGSizeMake(resolution.width, resolution.height) withParentGraph:loadedRoot];
+            CCNode * joint = [CCBDictionaryReader nodeGraphFromNodeGraphData:jointDict parentSize:CGSizeMake(resolution.width, resolution
+                    .height)                                 withParentGraph:loadedRoot];
```
Comment (rkachowski) [inline SpriteBuilder/ccBuilder/AppDelegate.m]: the height needs to be higher

File: SpriteBuilder/SpriteBuilder Tests/CCBDictionaryReader_Tests.m (hunk 1 above)
Comment (NickyWeber) [inline SpriteBuilder/SpriteBuilder Tests/CCBDictionaryReader_Tests.m:528]: Mumblings to myself, will be removed.

File: SpriteBuilder/ccBuilder/AppDelegate.m (hunk 2 above)
Comment (NickyWeber) [inline SpriteBuilder/ccBuilder/AppDelegate.m]: Appcode was abit too eager.

**Pull request merged**